src/Jobs/ClearIndexJob.php

```diff
@@ -14,31 +16,22 @@
 use SilverStripe\SearchService\Service\Traits\ServiceAware;
 use Symbiote\QueuedJobs\Services\AbstractQueuedJob;
 use Symbiote\QueuedJobs\Services\QueuedJob;
-use RuntimeException;
-use InvalidArgumentException;
 
-/**
- * Class ClearIndexJob
- * @package SilverStripe\SearchService\Jobs
- *
- * @property string $indexName
- * @property int $batchSize
- * @property int $batchOffset
- */
 class ClearIndexJob extends AbstractQueuedJob implements QueuedJob
 {
     use Injectable;
     use ServiceAware;
 
+    protected ?int $batchOffset = null;
```

Comment:
These should be `private` unless there's a specific need to `protected`, as it's better to make the API surface as small as possible (protected counts as API surface).  We can always change this from `private` to `protected` later if there's a need and release a new minor version

Reply:
Ok, all good. I made them protected thinking we might want to add the API, but I'm equally happy to **not** do that.

Reply:
@emteknetnz actually just a quick follow up question. Do we care that this is technically not backwards compatible if these properties aren't all `public` (since that's what the current visibility would be).

Reply:
Ug. Gosh these dynamic properties are awful.  They're deprecated though so they need to go regardless

Do we care? Sort of maybe ... they're not strictly API, though if another system relies on them we've broken backwards compat

Are there any existing magic methods _get() / _set() that map correctly? If not add is it feasible to add your own get/set methods?  You'll need to use public set* methods for this to work.

Reply:
All things considered, I'd say I'm happy to go with `private` visibility (and `protected` in the odd case where it's required by the module's own requirements)

Reply:
> That said, if you think there's basically no chance of other systems calling (read: project code) then I'm happy/I'd prefer that we don't worry about backwards compat and just use `private`

Walking a thin line here.. I know this isn't probably a widely adopted module yet, but sometimes we argue to death (little exaggerating here) to not break the tiniest thing from the semver/BC perspective and then we do anyway in some other module. I'm happy this work was done and appreciate everyone's effort, but couldn't spare the 'observation'. ✌️

Reply:
Fair point @michalkleiner

In the end we we went non-bc simply because we basically cannot transition off dynamic properties without a new major version of silverstripe-queuedjobs

This was the first time I've seen dynamic properties come up as a discussion point, so I'd say it's up in the air if this is considered API or not. They will very soon be a deprecated part of PHP so the aim should to remove their use.

Reply:
We are still BC at the moment. I had to revert back to dynamic properties in order to trigger the `__set()` method for job data.

Yes, I did add some `getters()` and `setters()`, but existing implementations should still be able to directly access those dynamic properties.

https://github.com/silverstripe/silverstripe-search-service/pull/69/commits/a35cc5b928d5353df1f8f6ecc9fcd8f1d79905eb